### ISSUE TITLE
Cannot save images locally when tag is too long

### ISSUE THREAD

**Issue by dchw** [earthly/earthly#2053]
Take, for example, this `Earthfile`:

```earthly
VERSION 0.6

long-tag:
    FROM alpine:3.16
    RUN echo "hi" > test
    SAVE IMAGE docker.io/this-is-a-super-long-company-name/and-a-super-duper-long-image-name:also-a-very-long-long-long-long-tag-for-reals
```

It fails with this error:
```
Error: build target: build main: failed to solve: image pull: 1 error occurred:
        * command failed: docker pull 127.0.0.1:8371/sess-3jrbwgudg3htkxk6z1rz8xghs/pullping:dockeriothisisasuperlongcompanynameandasuperduperlongimagenamealsoaverylonglonglonglongtagforreals-nYUCXTV4HWTYmiMRYFFl9m1yJj9beNnQ: exit status 1: invalid reference format: exit status 1

```
(https://ci.earthly.dev/logs?logId=a75aa13b-6caf-4a7c-b40b-44bf0c8010ba)

It appears that the additional metadata added to the image tag for the internal registry creates a tag that is too long.

It is possible to workaround this failure with `--no-output`, or by shortening your tags; but we should have a better way to handle this that doesn't depend on unbounded input from the user.

**Comment by dchw** [comment]
It has been observed by members of our community that this appears to be a regression in `v0.6.20`, as this appears to work in `v0.6.19`.

**Comment by dchw** [comment]
After futher investigation, this regressions was introduced in [a059067d8f5be7936c06c2bdad2cf0e9d6417b96](https://github.com/earthly/earthly/pull/1982)

**Comment by kclejeune** [comment]
commenting to mention that I'm encountering this issue as well when pushing images that are tagged with long branch names

**Comment by vyas-n** [comment]
Hi folks,

I also ran into the same issue using Docker Desktop for macOS with Earthly.
For me, it was this setting that caused the error to appear:

<img width="784" alt="Screen Shot 2022-09-18 at 01 56 14" src="https://user-images.githubusercontent.com/86535092/190889922-f551ee06-f5c1-430d-9033-5840857793dc.png">

When enabled, it seems to mess with the container registry settings that are incompatible with any recent version of Earthly (I tested with 0.6.14, 0.6.19, & 0.6.22).

Disabling the setting fixed it for me.
